### ISSUE TITLE
Add cause assertions to ThrowableAssert

### ISSUE THREAD

**Issue by ash2k** [alexruiz/fest-assert-2.x#112]
Add two new assertions to `ThrowableAssert`:

``` java
hasCause(Throwable e);
hasRootCause(Throwable e);
```

Also, maybe (is this too specific?) add `instanceOf` checks for cause:

``` java
hasCauseOfClass(Class<? extends Throwable> eClass);
hasRootCauseOfClass(Class<? extends Throwable> eClass);
```

**Comment by alexruiz** [comment]
I'll be removing ThrowableAssert from the code base. It provide little benefit since anything that you need to check in a Throwable can be done with other assertions.

**Comment by alexruiz** [comment]
Pressed "Close" by accident. I'm leaving the issue open in case we'd like to discuss it further.

**Comment by ash2k** [comment]
From wiki:

```
FEST Assertions 2.0 is a Java library that provides a fluent interface for writing assertions.
Its main goal is to improve test code readability and make maintenance of tests easier.
```

``` java
assertThat(throwable).hasRootCause(e);
```

``` java
assertThat(Throwables.getRootCause(throwable)).isEqualTo(e);
```

Dont you agree that the first sample is MUCH more radable?

[Throwables](http://docs.guava-libraries.googlecode.com/git-history/release/javadoc/com/google/common/base/Throwables.html) is from Guava.

**Comment by alexruiz** [comment]
It is more readable, no doubt. But we cannot provide an assertion for everything. We need to draw the line somewhere. We may reconsider after releasing 2.0.

**Comment by joel-costigliola** [comment]
I agree with Mikhail, the thing that made me choose Fest over Hamcrest was the richness of Fest Assertions, IMHO this is why Fest is successfull (along with the ease of use).

Question is : do we had some user complain about providing too much assertions ?

I'm gonna post the question on the fest-dev list of where we should draw the line of what we provide.
Maybe the solution would a small core assertions and an extended assertions module ... but let's disuss that on the dev-list.

**Comment by alexruiz** [comment]
Good idea, Joel! I think automatic notification of issues (creation, updates) to the dev mailing list can be pretty helpful. This way, others can participate.
